interview-questions.md

```diff
@@ -9,59 +9,59 @@ Then, research the question to expand on your answer. Even if you feel you have
 
 1. What is `this` in JavaScript?
 
-  Your answer:
+  Your answer:It Refers to an object that it belongs to. This is placed inside of the code block of the component and has a key value.
 
-  Researched answer:
+  Researched answer: My answer was sort of correct in the fact that it refers to an object. React is made up of two components types the first component type is the one that holds state and the other one doesn't hold a state. So therefore the keyword `this` helps to go in and out of components that hold his state and doesn't hold a state. A state is an object that stores a components dynamic data and determines how the component renders and behaves.
 
 
 
 2. What is yarn? What file(s) are modified in a React application when you run yarn?
 
-  Your answer:
+  Your answer: Yarn is a package manager for your code it makes it easier to share your code with other people. It gives you three additional files Node modules, yarn Lock and package.json.
 
-  Researched answer:
+  Researched answer: So I was right yarn is a package manager and it does make it easier for anybody in the world to read your code it also allows you to use other developers solutions to different problems making it easier for you to develop your software. So that is why it Modifies the files in your react application so that I can guarantee a smooth transition from one developer to the other because we all use different text editors.
 
 
 
 3. What are props in React?
 
-  Your answer:
+  Your answer: Props are things in react that you can actually touch and move around and do things with.They are like a menu item with out props all it is is hardcoded in and props allows you to manipulate and click on as a user.
 
-  Researched answer:
+  Researched answer: Props are how are you pass data from one component to the other as parameters.Props are also like arguments passed into react components they also act like HTML attributes.If Your components have a constructior function the props should always be passed through the constructor also to the react component via the super method and as we know the super method allows you to inherit to the parent.
```

Comment:
Yea, props are data. Data that is accessible between files because of how the flow of data is set up.